core/src/main/java/com/axellience/vuegwt/core/client/component/VueComponent.java

```diff
@@ -98,7 +98,41 @@
 
   public native void $off(String name, OnEvent callback);
 
-  public native void $emit(String name, @DoNotAutobox Object... param);
+  @JsOverlay
+  public final void $emit(String name, int param) {
+    $emit(name, Js.asAny(param));
+  }
+  @JsOverlay
+  public final void $emit(String name, float param) {
+    $emit(name, Js.asAny(param));
+  }
+  @JsOverlay
+  public final void $emit(String name, double param) {
+    $emit(name, Js.asAny(param));
+  }
+  @JsOverlay
+  public final void $emit(String name, short param) {
+    $emit(name, Js.asAny(param));
+  }
+  @JsOverlay
+  public final void $emit(String name, boolean param) {
+    $emit(name, Js.asAny(param));
+  }
+  @JsOverlay
+  public final void $emit(String name, byte param) {
+    $emit(name, Js.asAny(param));
+  }
+  @JsOverlay
+  public final void $emit(String name, char param) {
+    $emit(name, Js.asAny(param));
+  }
+  @JsOverlay
+  public final void $emit(String name, Object param) {
+    $emit(name, Js.asAny(param));
+  }
+
+  public native void $emit(String name, Any param);
+  public native void $emit(String name);
```

Comment:
The original intent of the varargs seemed to be to make this value optional. I might be wrong though.

Reply:
It wasn't exactly the intent, you can emit an event with several arguments and the receiver can receive them all. However this is not a documented behavior in `Vue.js`, so Vue GWT should probably not rely on it. This will be a breaking change for some users of the library though (but the workaround is easy, create an object to combine what you used to pass as 2 params).